controller/konnect/constraints/constraints.go

```diff
@@ -75,3 +75,14 @@ type SupportedKonnectEntityPluginReferenceableType interface {
 type EntityWithKonnectAPIAuthConfigurationRef interface {
 	GetKonnectAPIAuthConfigurationRef() konnectv1alpha1.KonnectAPIAuthConfigurationRef
 }
+
+// ClientObject represents an object that has a name and a namespace
+// (might be empty for cluster-scoped objects).
+type ClientObject[T any] interface {
+	*T
+
+	GetName() string
+	GetNamespace() string
+
+	client.Object
```

Comment:
Doesn't `client.Object` already provide `GetName` and `GetNamespace`?

Reply:
BTW Wouldn't it be possible to omit this interface and use the existing `EntityType`?

Reply:
Ah yes 🤦 27e38911dad8aee4c5b3fbab88d93c5900742655

Reply:
I agree it's less restrictive, but we only use it now for Konnect entities (namely, in `enqueueObjectForKonnectGatewayControlPlane`, which implies a Konnect entity use case). We might as well keep it simple, not introduce more types, and add it when we find it required.

Reply:
👍 f382e23f8fd1547593f6371f3b25777e01746034